lms/services/canvas.py

```diff
@@ -1,14 +1,47 @@
 from lms.services.exceptions import CanvasAPIPermissionError, CanvasFileNotFoundInCourse
 
 
+def is_speedgrader(request):
+    if "learner_canvas_user_id" in request.GET:
+        # Location while launching an assignment
+        return bool(request.GET["learner_canvas_user_id"])
+
+    # Location when using the Sync API
+    try:
+        return "learner" in request.json
+    except:  # pylint:disable=bare-except
+        return False
+
+
+def group_set(request):
+    if "group_set" in request.params:
+        # Location while launching an assignment
+        value = request.params["group_set"]
+    else:
+        # Location when using the Sync API
+        value = request.json["learner"].get("group_set")
+
+    return int(value)
+
+
+def is_canvas_group_launch(request):
+    try:
+        group_set(request)
+    except (KeyError, ValueError, TypeError):
+        return False
+    else:
+        return True
+
```

Comment:
I having trouble finding the right place for these.

- Adding them to the existing `CanvasService` means that we'd need `CanvasService` in `resource/lti_resource` and JSConfig. That also brings `canvas_api_client` which needs a applications insntace (via `get_current`). 

This causes some functests to fail as just using the resources requires a application instance which was not the case before.

- Using a different service class. I think that will mean we have CanvasService, CanvasAPIClient + a new one. It can work but not sure about the naming for that one. `CanvasHelpers`, `CanvasRequest`, meh.

- Using plain function as in this diff. Works as well. It needs a bit more importing in general and individual patching in the tests.